common/new_process.py

```diff
@@ -71,9 +72,7 @@ def _mirror_output(process: subprocess.Popen, output_files: List) -> str:
     while True:
         # See if we can get a line from the queue.
         try:
-            # TODO(metzman): Handle cases where the process does not have utf-8
-            # encoded output.
-            line = out_queue.get_nowait().decode('utf-8', errors='ignore')
+            line = out_queue.get(timeout=WAIT_SECONDS).decode('utf-8', errors='ignore')
```

Comment:
Should we sleep at end of loop (while True) using time.sleep. That way, we dont have a forced 5 sec delay for processes that immediately exit (trigger queue.empty and then exit from thread.is_alive) ?

Reply:
adding @oliverchang for review as well, and how he thinks on this approach ?

Reply:
As discussed offline, this happens to be fine because runner.py is the only caller that reaches this. I'm going to refactor this tomorrow to not rely on this

Reply:
Seems like either approach (sleep vs timeout here) would be prone to a race. Refactor SGTM.